### ISSUE TITLE
View command to open DataFrame in Variable explorer

### ISSUE THREAD

**Issue by Diego-MX** [spyder-ide/spyder#5137]
## Description

I come from R with RStudio and I'd like to integrate more Python in my workflow.  
With Spyder, I have been able to make this easy.  What I've found in Spyder + pandas is really cool and has helped me a bunch. 
One feature that I think is very comfortable from my previous workbench is the `View` command, which directly opens a DataFrame in the Variable Explorer.   Thus allowing you to keep the hands in the keyboard.  I didn't find a similar command here, so I'm taking the liberty to suggest it. 
Cheers. 

## Versions and main components

* Spyder Version: 3.2.1
* Python Version:  3.6.1
* Qt Version:  5.6.2
* PyQt Version: 5.6
* Operating system:  Linux-Ubuntu16.04

## Dependencies

IPython >=4.0    :  5.3.0 (OK)
cython >=0.21    :  0.25.2 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
numpy >=1.7      :  1.12.1 (OK)
pandas >=0.13.1  :  0.20.1 (OK)
psutil >=0.3     :  5.2.2 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pyflakes >=0.6.0 :  1.5.0 (OK)
pygments >=2.0   :  2.2.0 (OK)
pylint >=0.25    :  1.6.4 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
sphinx >=0.6.6   :  1.5.6 (OK)
sympy >=0.7.3    :  1.0 (OK)

**Comment by goanpeca** [comment]
@Diego-MX could you describe the actual workflow, or how this command works?

**Comment by ccordoba12** [comment]
I think @Diego-MX wants to write something like

    view(df)

in a console and make it display a Dataframe directly, instead of having to double click on the Variable Explorer.

**Comment by Diego-MX** [comment]
I have edited my question.  Thank you.

**Comment by goanpeca** [comment]
@ccordoba12 what do you think? I guess we could have, spyder magics for ipython?

**Comment by ccordoba12** [comment]
We already have one called `%varexp` to create plots from the Variable Explorer. We could extend it to provide something like this.

**Comment by goanpeca** [comment]
>We already have one called %varexp to create plots from the Variable Explorer. We could extend it to provide something like this.

Sounds good. I think a magic is better suited for this right?

**Comment by ccordoba12** [comment]
Yep, it's the right concept to use.

**Comment by tjhallum** [comment]
I would very much like a feature like the one described above by Diego-MX. Was a feature like this ever implemented?

**Comment by ccordoba12** [comment]
No, sorry. We plan to implement it in Spyder 5, to be released in 2020.

**Comment by tjhallum** [comment]
@ccordoba12 Sounds good, thank you for the expedient reply!

**Comment by akdor1154** [comment]
another use case: would be make it easier to view a subset of a large dataframe without having to create a temporary variable.
e.g. `explore(df['TheColumnICareAbout', 'AnotherColumn'])`

**Comment by ccordoba12** [comment]
@akdor1154, that won't be possible.

@impact27, I think this should be very easy to implement by using the new comms architecture and adding a `%view` magic to our kernel.

**Comment by akdor1154** [comment]
@ccordoba12, is there an architecture where that would be possible? e.g. IPython's `display()` achieves this workflow in notebooks, and it's really handy.

**Comment by ccordoba12** [comment]
Then you're welcome to dive in our architecture and implement it, if you think it's possible.

**Comment by impact27** [comment]
I think the best would be to have a spyder namespace for spyder magic to cleanly separate from ipython magic. something like `%spyder display df` or `%spyder.display df`.

What I would see is:
In spyder:
 - in `spyder/plugins/variableexplorer`, define a new function to display a variable that is not in the variable explorer (for @akdor1154 suggestion) Or that finds the index of the object.
 - in `spyder/plugins/ipythonconsole/widgets/shell.py`: register a new `call_handler`: eg. `'display_variable': self.display_variable` that will send the received object (or object name if only the index is needed) to the variable explorer.
        
In spyder-kernels:
 - Register the magic with ipython `MagicsManager.register_function`
 - Send the data to be displayed to the frontend with a comm message to `display_variable`

**Comment by papergr** [comment]
> another use case: would be make it easier to view a subset of a large dataframe without having to create a temporary variable.
> e.g. `explore(df['TheColumnICareAbout', 'AnotherColumn'])`

As a previously RStudio user transiting to Spyder this would be extremely useful; keyboard use only plus record-wise filtering saves you in cases of significantly sized variables where variable explorer window cannot open.

**Comment by carrickpaul** [comment]
Was this implemented?

**Comment by CTimmerman** [comment]
[It has](https://stackoverflow.com/questions/60097076/view-dataframe-while-debugging-in-vs-code) in Visual Studio Code, which is [open source (MIT license)](https://github.com/microsoft/vscode) so that code might be useful in Spyder.

**Comment by ccordoba12** [comment]
@CTimmerman, I don't see anything related to this in the StackOverflow question you referenced above. This is about adding a command to the IPyhon console that allows to view variables directly in Spyder.

**Comment by CTimmerman** [comment]
https://stackoverflow.com/a/65389490/819417 says you can just enter the name in the debug console there.

**Comment by ccordoba12** [comment]
You can do the same in Spyder but it'll give you a text-based representation of variables. What others are asking for here is the equivalent of the `view` command in R:

https://www.statology.org/view-function-in-r/
